src/BlockHeader.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {
+  getLabel,
+  getNestedBlockDefinition, isNA, isSimpleLayout,
+  isStruct, structValueToObject, triggerCustomEvent
+} from './processing/utils';
+import {toggleBlock} from './actions';
+import BlockActions from './BlockActions';
+import {refType} from './types';
+
+
+@connect((state, props) => {
+  const {fieldId, blockId} = props;
+  const fieldData = state[fieldId];
+  const blocks = fieldData.blocks;
+  const block = blocks[blockId];
+  const blockDefinition = getNestedBlockDefinition(state, fieldId, blockId);
+  const value = block.value;
+  return {
+    isSimpleLayout: isSimpleLayout(blockDefinition, fieldData.isMobile),
+    blockDefinition,
+    icons: fieldData.icons,
+    value: isStruct(blockDefinition) ?
+             structValueToObject(state, fieldId, value)
+             :
+             value,
+  };
+}, (dispatch, props) => {
+  const {fieldId, blockId} = props;
+  return bindActionCreators({
+    toggleBlock: () => toggleBlock(fieldId, blockId),
+  }, dispatch);
+})
```

Comment:
It seems to me like this component could receive its data from `Block`, instead of also be connected to Redux.